tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+
+                # Fetching the acting set for the pool
+                # Getting a sample PG from the pool
+                pool_id = pool_obj.get_pool_id(pool_name=pool)
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+
+                log.debug(f"Acting set for PG-id {pgid} in pool {pool} is {acting_set}")
+
+                # Trying to change the primary with existing primary. -ve test
+                new_osd = acting_set[0]
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = (
+                    f"Error EINVAL: osd.{new_osd} is already primary for pg {pgid}"
+                )
+
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.debug(
+                        "Could not change the primary with the existing primary. Pass"
+                    )
+                    log.error(f"An error occurred but was expected: {e}")
+
+                # Trying to set a non-existant OSD as for the PG primary. -ve test
+                new_osd = 250
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = f"Error ENOENT: osd.{new_osd} does not exist"
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
+                        )
+                        raise Exception("Wrong error string obtained")
+                    log.error(f"An error occurred but was expected: {e}")
+                    log.debug(
+                        "Could not change the primary with non existent OSD. Pass"
+                    )
+
+                # Trying to set a OSD which is not present in the acting set. -ve test
+                osd_list = []
+                for node in ceph_cluster:
+                    if node.role == "osd":
+                        node_osds = rados_obj.collect_osd_daemon_ids(node)
+                        osd_list = osd_list + node_osds
+
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                for item in osd_list:
+                    if item not in acting_set:
+                        new_osd = item
```

Comment:
maybe we should break from this for loop when we are getting an OSD which is not part of acting set, no point in continuing the loop as the purpose is fulfilled

Reply:
Done.